### PR TITLE
fix(@desktop/chat): sign and send appears for both recipient and sender when sharing the address

### PR DESCRIPTION
**Pull request by saledjenic**
The reason for this issue is a message where recipient accepted to share his address with sender.
In that message recipient's public key is set as a "from" property of a "Message" object and we
cannot determine which of two users has initiated transaction actually.

This is fixed checking if the "from" address from the "commandParameters" object of the "Message"
is contained as an address in the wallet of logged in user. If yes, means that currently logged in
user has initiated a transaction (he is a sender), otherwise currently logged in user is a
recipient.

We were just sending a transaction, without notifying message about that. Now we call
callPrivateRPC("acceptRequestTransaction".prefix, %* [transactionHash, messageId, signature])
and that notifies message about the change, but only on the sender side. Appropriate message
on the recipient side was not notified about the change. That need to be checked.

Fixes: #2719

### CODE REVIEW

File: src/status/signals/messages.nim [hunk 1]
```diff
@@ -362,7 +369,36 @@ proc toMessage*(jsonMsg: JsonNode, pk: string): Message =
...
+    # This is kind of a workaround in case we're processing a transaction message. The reason for 
+    # that is a message where a recipient accepted to share his address with sender. In that message
+    # a recipient's public key is set as a "from" property of a "Message" object and we cannot 
+    # determine which of two users has initiated transaction actually. 
+    # 
+    # To overcome this we're checking if the "from" address from the "commandParameters" object of 
+    # the "Message" is contained as an address in the wallet of logged in user. If yes, means that
+    # currently logged in user has initiated a transaction (he is a sender), otherwise currently 
+    # logged in user is a recipient.
+    message.isCurrentUser = currentUserWalletContainsAddress(message.commandParameters.fromAddress)
```
Comment (jrainville) [inline src/status/signals/messages.nim:381]: Maybe we could add an `if` to check if there is a `message.commandParameters.fromAddress` first, that way we don't call `currentUserWalletContainsAddress` with an empty sring.

File: src/status/signals/messages.nim [hunk 2]
```diff
@@ -362,7 +369,36 @@ proc toMessage*(jsonMsg: JsonNode, pk: string): Message =
...
+    echo "a"
+    echo "a"
+    echo "a"
+    echo "ADDRESS: ", message.commandParameters.fromAddress, "  containedInLocalWallet: ", message.isCurrentUser
+    echo "a"
+    echo "a"
+    echo "a"
+    echo "a"
+    echo "a"
+    echo "a"
```
Comment (jrainville) [inline src/status/signals/messages.nim:395]: A "little" bit of left over here haha

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/TransactionBubble.qml [hunk 3]
```diff
@@ -207,6 +202,44 @@ Item {
...
+
+                let transactionId = response.result
+
+                if (!response.success) {
+                    if (Utils.isInvalidPasswordMessage(transactionId)){
+                        //% "Wrong password"
+                        transactionSigner.validationError = qsTrId("wrong-password")
+                        return
+                    }
+                    sendingError.text = transactionId
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/TransactionBubble.qml:223]: I think those errors won't work now that you moved the Connectin in the bubble.

**Review by jrainville — COMMENTED** [review]
Looks very good. Nice job.

I just have 2 points to check first

**Comment by saledjenic** [comment]
@jrainville I removed that logging code which left over, definitely I overlooked it when I committed changes.

> Maybe we could add an if to check if there is a message.commandParameters.fromAddress first, that way we don't call currentUserWalletContainsAddress with an empty sring.
Actually better place to add that check is currentUserWalletContainsAddress proc, cause that proc may be called later from multiple places and I added it there.

> I think those errors won't work now that you moved the Connectin in the bubble.
It will work everywhere as long as you connect the same valid target (walletModel.transactionsView) to appropriate slot.

**Comment by jrainville** [comment]
> It will work everywhere as long as you connect the same valid target (walletModel.transactionsView) to appropriate slot.

Sorry, I think I wasn't clear. What I meant is that the `sendingError` and `transactionSigner` components are probably not accessible since you moved the code to the TransactionBubble. Since the components are in the `SignTransactionModal`, it will probably fail.

**Comment by jrainville** [comment]
This probably fixes #2413 as well

**Comment by saledjenic** [comment]
> > It will work everywhere as long as you connect the same valid target (walletModel.transactionsView) to appropriate slot.
> 
> Sorry, I think I wasn't clear. What I meant is that the `sendingError` and `transactionSigner` components are probably not accessible since you moved the code to the TransactionBubble. Since the components are in the `SignTransactionModal`, it will probably fail.

Yes, you're right. But another important thing I overlooked is that SignTransactionModal component is used not only from TransactionBubble, but also BrowserLayout and AcceptTransaction and moving connection from the place where it was initially corrupts behaviour in those two components. So I moved it back and updated accordingly.

**Review by jrainville — APPROVED** [review]

**Pull request merged**